qiita_pet/handlers/study_handlers/listing_handlers.py

```diff
@@ -42,14 +43,118 @@ def _get_shared_links_for_study(study):
+    if info['pmid'] is not None:
+        info['pmid'] = ", ".join([pubmed_linkifier([p])
+                                  for p in info['pmid']])
+    else:
+        info['pmid'] = ""
+    if info["number_samples_collected"] is None:
+        info["number_samples_collected"] = 0
+    info["shared"] = _get_shared_links_for_study(study)
+    info["num_raw_data"] = len(study.raw_data())
+    info["status"] = status
+    info["study_id"] = study.id
+    info["pi"] = study_person_linkifier((PI.email, PI.name))
+    del info["principal_investigator_id"]
+    del info["email"]
+    # Build the proc data info list for the child row in datatable
+    info["proc_data_info"] = _build_single_proc_data_info(
+        study, study_proc, proc_samples)
+    return info
+
+
+def _build_single_proc_data_info(study, study_proc, proc_samples):
+    """Build the proc data info list for the child row in datatable
+
+
+    Parameters
+    ----------
+    study : Study object
+        The study to build information for
+    study_proc : dict of lists
+        The processed data attached to he study, in the form
+        {study_id: [proc_data_id, proc_data_id, ...], ...}
+    proc_samples : dict of lists
+        The samples available in the processed data, in the form
+        {proc_data_id: [samp1, samp2, ...], ...}
+
+    Returns
+    -------
+    dict of dicts
+        The information for the processed data, in the form
+        {proc_data_id: {info: value, ...}, ...}
+    """
+    proc_data_info = []
+    for pid in study_proc[study.id]:
+        proc_data = ProcessedData(pid)
+        proc_info = proc_data.processing_info
+        proc_info['pid'] = pid
+        proc_info['data_type'] = proc_data.data_type()
+        proc_info['samples'] = sorted(proc_samples[pid])
+        proc_info['processed_date'] = str(proc_info['processed_date'])
+        proc_data_info.append(proc_info)
+    return proc_data_info
+
+
+def _build_study_info(user, study_proc=None, proc_samples=None):
+    """builds list of dicts for studies table, with all html formatted
+
+    Parameters
+    ----------
+    user : User object
+        logged in user
+    study_proc : dict of lists, optional
+        Dictionary keyed on study_id that lists all processed data associated
+        with that study. Required if proc_samples given.
+    proc_samples : dict of lists, optional
+        Dictionary keyed on proc_data_id that lists all samples associated with
+        that processed data. Required if study_proc given.
+
+    Returns
+    -------
+    infolist: list of dict of lists and dicts
+        study and processed data info for JSON serialiation for datatables
+        Each dict in the list is a single study, and contains the text
+
+    Notes
+    -----
+    Both study_proc and proc_samples must be passed, or neither passed.
+    """
+    build_samples = False
+    # Logic check to make sure both needed parts passed
+    if study_proc is not None and proc_samples is None:
```

Comment:
I think there's no tests for these exception cases. Would you mind adding those?